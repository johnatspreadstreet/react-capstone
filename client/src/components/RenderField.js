import React from 'react'

// type can be: text, password, submit, reset, radio, checkbox, button, 
// type can be: color, date, datetime-local, email, month, number, range, search, tel, time, url, week

const RenderField = ({ options, input, label, type, meta: { touched, error } }) => {
    return (type === 'text' || type === 'email' || type === 'number' || type === 'checkbox') ?
    (
        <div>
          <label className='label-input'>{label}</label>
          <div>
            <input {...input} type={type} placeholder={label} className='field-input' />
            {touched && error && <span>{error}</span>}
          </div>
        </div>
    ) : (type === 'select') ?
    (
        <div>
          <label className='label-input'>{label}</label>
          <div>
            <select {...input} type={type} placeholder={label}>
                {options.map((option, index) => {
                    return <option key={index} value={option}>{option}</option>
                })}
            </select>
            {touched && error && <span>{error}</span>}
          </div>
        </div>
    )
    : <div>Not a correct type</div>
}

export default RenderField;