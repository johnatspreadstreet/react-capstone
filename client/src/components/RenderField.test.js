import React from "react";
import { shallow, mount } from "enzyme";
import RenderField from "./RenderField";

describe("A suite of tests for the RenderField component", () => {
  // Smoke test
  it("should render Redux Form specific components with default props", () => {
    const component = shallow(
      <RenderField meta={{ touched: true, error: null }} />
    );

    expect(component).toMatchSnapshot();
  });
});
