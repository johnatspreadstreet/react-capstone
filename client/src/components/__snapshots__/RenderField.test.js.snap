// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite of tests for the RenderField component should render Redux Form specific components with default props 1`] = `
<div>
  <label
    className="label-input"
  />
  <div>
    <input
      className="field-input"
    />
  </div>
</div>
`;

exports[`A suite of tests for the RenderField component should render Redux Form specific components with no props 1`] = `
<div>
  <label
    className="label-input"
  />
  <div>
    <input
      className="field-input"
    />
  </div>
</div>
`;
